### DISCUSSION TITLE
PreferApply config option not always work

### DISCUSSION THREAD

**Discussion by plukawski** [linq2db/linq2db#3836] [Q&A]
I have noticed that setting **Configuration.Linq.PreferApply** to false does not always prevent CROSS APPLY/OUTER APPLY generation instead of regular joins in final SQL. Sometimes this affects performance in a bad way. I have noticed that since the introduction of that configuration option the following line was always commented out:
https://github.com/linq2db/linq2db/blob/f939b6b8594d30fd127ba581efaaaa9a5470a817/Source/LinqToDB/Linq/Builder/SelectManyBuilder.cs#L128

Is this a mistake that this line is commented out since the very beginning? In my opinion setting the **PreferApply** option to **false** should prevent CROSS APPLY/OUTER APPLY from being generated for joining logic except from the cases where it is required by the query logic explicitly.

**Comment by MaceWindu** [comment]
@sdanyliv

**Reply by sdanyliv** [reply]
In new translator this code rewritten. It will predict better when to use CROSS/OUTER APPLY join. In current realization it is complicated task.

**Comment by plukawski** [comment]
@sdanyliv , so correct me if I understood you correctly:
1. There is undergoing refactor of the linq to SQL language translator used in Linq2Db.
2. In current design implementing the mentioned configuration option is a complicated task, so it will be done during the refactor in the new translator.

Is my understanding correct? Is there defined any ETA of the new Translator that supports this option then?
